### PR TITLE
Fix a bug when restoring the layouts from the db

### PR DESCRIPTION
**Pull request by cquiroz**
This fixes a bug restoring the obs layout coming from the user preferences db

Signed-off-by: Carlos Quiroz <carlos.quiroz@noirlab.edu>

### CODE REVIEW

File: common/src/main/scala/explore/components/TileController.scala [hunk 1]
```diff
@@ -104,9 +106,13 @@ object TileController {
         getBreakpointFromWidth(p.layoutMap.map { case (x, (w, _, _)) => x -> w }, p.gridWidth)
       }
       // Store the current layout
-      .useStateViewWithReuseBy((p, _, _) => updateResizableState(p.layoutMap))
+      .useStateViewBy((p, _, _) => updateResizableState(p.layoutMap))
+      // Update the current layout if it changes upstream
+      .useEffectWithDepsBy((p, _, _, _) => p.layoutMap)((_, _, _, current) =>
```
Comment (cquiroz) [inline common/src/main/scala/explore/components/TileController.scala:111]: This is the core of the fix.

File: common/src/main/scala/explore/common/UserPreferencesQueries.scala [hunk 2]
```diff
@@ -14,14 +15,16 @@ import explore.model.ResizableSection
 import explore.model.enums.PlotRange
 import explore.model.enums.TimeDisplay
 import explore.model.enums.Visible
-import explore.model.layout._
+import explore.model.layout.{_, given}
```
Comment (rpiaggio) [inline common/src/main/scala/explore/common/UserPreferencesQueries.scala:18]: I think Scala 2 style `import X._` imports givens already. Scala 3 style would be `import X.{*, given}`.

File: common/src/main/scala/explore/components/TileController.scala [hunk 3]
```diff
@@ -16,7 +18,7 @@ import explore.implicits._
 import explore.model.Constants
 import explore.model.GridLayoutSection
 import explore.model.enums.TileSizeState
-import explore.model.layout._
```
Comment (rpiaggio) [inline common/src/main/scala/explore/components/TileController.scala:19]: `import *`

File: common/src/main/scala/explore/components/TileController.scala (hunk 1 above)
Comment (rpiaggio) [reply common/src/main/scala/explore/components/TileController.scala:111]: Thanks. I wonder if we should implement a custom hook that implements the `useState`/`useEffectWithDeps` pattern.

**Review by rpiaggio — APPROVED** [review]
LGTM

File: common/src/main/scala/explore/common/UserPreferencesQueries.scala (hunk 2 above)
Comment (cquiroz) [reply common/src/main/scala/explore/common/UserPreferencesQueries.scala:18]: Yes, scalafmt for some reason converts them to `{_, given}` I guess we need two lines of imports

Comment (cquiroz) [reply common/src/main/scala/explore/common/UserPreferencesQueries.scala:18]: or maybe `OrganizeImports`

**Pull request merged**